vnext/Microsoft.ReactNative/Modules/I18nManagerModule.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "I18nManagerModule.h"
+#include <IReactDispatcher.h>
+#include <winrt/Windows.ApplicationModel.Resources.Core.h>
+#include <winrt/Windows.Globalization.h>
+#include "Unicode.h"
+
+namespace Microsoft::ReactNative {
+
+static const React::ReactPropertyId<bool> &SystemIsRTLPropertyId() noexcept {
+  static const React::ReactPropertyId<bool> prop{L"I18n", L"SystemIsRTL"};
+  return prop;
+}
+
+static const React::ReactPropertyId<bool> &AllowRTLPropertyId() noexcept {
+  static const React::ReactPropertyId<bool> prop{L"I18n", L"AllowRTL"};
+  return prop;
+}
+
+static const React::ReactPropertyId<bool> &ForceRTLPropertyId() noexcept {
+  static const React::ReactPropertyId<bool> prop{L"I18n", L"ForceRTL"};
+  return prop;
+}
+
+void I18nManager::InitI18nInfo(const winrt::Microsoft::ReactNative::ReactPropertyBag &propertyBag) noexcept {
+  auto layoutDirection =
+      winrt::Windows::ApplicationModel::Resources::Core::ResourceContext().GetForCurrentView().QualifierValues().Lookup(
+          L"LayoutDirection");
+
+  propertyBag.Set(SystemIsRTLPropertyId(), layoutDirection != L"LTR");
+}
+
+/*static*/ bool I18nManager::IsRTL(const React::ReactPropertyBag &propertyBag) noexcept {
+  if (propertyBag.Get(ForceRTLPropertyId()))
```

Comment:
>if (propertyBag.Get(ForceRTLPropertyId())) [](start = 2, length = 42)

Please note that this code is incorrect.
The PropertyBag returns std::optional<bool> and this code only checks if value is present, and not if it is true or false. The correct code should probably look like this:
```
  if (propertyBag.Get(ForceRTLPropertyId()).value_or(false))
    return true;
```
The same is for the other two cases below.

Alternatively, we may store only 'true' values in the property bag. In that case the code here would be the same, but the initialization must be changed to
```
m_context.Properties().Set(ForceRTLPropertyId(), forceRTL ? std::optional{forceRTL} : std::optional<bool>{});
```
Though, changing the condition is better in case if some other code decides to set these properties for us.